graph/src/blockchain/firehose_block_ingestor.rs

```diff
@@ -82,6 +82,87 @@ where
         }
     }
 
+    pub async fn run_backfill(self) {
+        use firehose::ForkStep::*;
+
+        let mut backoff =
+            ExponentialBackoff::new(Duration::from_millis(250), Duration::from_secs(30));
+
+        let mut backfill_cursor = self.fetch_backfill_cursor().await;
+
+        loop {
+            let backfill_completed = self.fetch_backfill_is_completed().await;
+            if backfill_completed {
+                info!(self.logger, "Backfill completed");
+                break;
+            }
+
+            let backfill_target_block_number = self.fetch_backfill_target_block_num().await;
```

Comment:
This should be done outside the main loop, there is no need to fetch the backfill target block num on every loop, only once.